great_expectations/data_context/store/tuple_store_backend.py

```diff
@@ -796,3 +796,144 @@ def remove_key(self, key):
     def _has_key(self, key):
         all_keys = self.list_keys()
         return key in all_keys
+
+
+class TupleAzureBlobStoreBackend(TupleStoreBackend):
+    """
+    Uses an Azure Blob as a store.
+
+    The key to this StoreBackend must be a tuple with fixed length based on the filepath_template,
+    or a variable-length tuple may be used and returned with an optional filepath_suffix (to be) added.
+    The filepath_template is a string template used to convert the key to a filepath.
+    
+    You need to setup the connection string environment variable 
+    https://docs.microsoft.com/en-us/azure/storage/blobs/storage-quickstart-blobs-python
+    """
+
+    # We will use blobclient here
+    def __init__(
+        self,
+        container,
+        prefix="",
+        filepath_template=None,
+        filepath_prefix=None,
+        filepath_suffix=None,
+        forbidden_substrings=None,
+        platform_specific_separator=False,
+        fixed_length_key=False,
+    ):
+        super().__init__(
+            filepath_template=filepath_template,
+            filepath_prefix=filepath_prefix,
+            filepath_suffix=filepath_suffix,
+            forbidden_substrings=forbidden_substrings,
+            platform_specific_separator=platform_specific_separator,
+            fixed_length_key=fixed_length_key,
+        )
+        self.prefix = prefix
+        self.container = container
+
+    def _get_container_client(self):
```

Comment:
Would it be possible to get rid of environment variable? If I get it right, the current version of _get_container_client limit the usage of store for just single Azure connection. Without that limitation it would be possible to add multiple connections, f.e. one blob storage for validation store and second one for docs etc.

Reply:
Hi @roman-novosad, thank you for the review. Yes, you are totally right. 

As an alternative we can put it inside the `great_expecation.yaml`, I am not sure about this alternative since connection string contains sensitive information. What do you think @eugmandel & @roman-novosad ?

Reply:
Don't worry, according to documentation, you can "Replace credentials or other values in your great_expectations.yml with ${}-wrapped variable names (i.e. ${ENVIRONMENT_VARIABLE} or ${YAML_KEY}).". So secrets stay secret.

Reply:
Thank you @roman-novosad, I am glad that you tell me about it. I change the connection_string mechanism to read it from great_expectation.yaml file.